### ISSUE TITLE
`npm:` specifiers

### ISSUE THREAD

**Issue by loynoir** [oven-sh/bun#2733]
### What is the problem this feature would solve?

`node_modules` is ugly

### What is the feature you are proposing to solve the problem?

https://deno.com/manual@v1.32.5/node/npm_specifiers

```ts
import chalk from "npm:chalk@5";

console.log(chalk.green("Hello!"));
```

**Comment by Jarred-Sumner** [comment]
In Bun, the `npm:` specifier is unnecessary.

Instead of
```js
import chalk from "npm:chalk@5";

console.log(chalk.green("Hello!"));
```

You can do 
```js
import chalk from "chalk";

console.log(chalk.green("Hello!"));
```

If you have no `node_modules` folder, try:
```js
import chalk from "chalk@5";

console.log(chalk.green("Hello!"));
```

We aren't going to support the `npm:` prefix.

**Comment by loynoir** [comment]
@Jarred-Sumner 

It does not work.

```sh
$ cat reproduce.mjs 
import chalk from 'chalk@5'

console.log(chalk.green('Hello!'))
$ bun reproduce.mjs 
1 | import chalk from 'chalk@5'
2 | 
3 | console.log(chalk.green('Hello!'))
               ^
TypeError: undefined is not an object (evaluating 'chalk.green')
      at /path/to/reproduce.mjs:3:12
```

**Comment by Jarred-Sumner** [comment]
Is this on Bun v0.5.9 (or canary)?

I'm having trouble reproducing the issue you're seeing

<img width="365" alt="image" src="https://user-images.githubusercontent.com/709451/234230784-4b87602d-6dd3-4338-b216-b3f33097e262.png">

**Comment by loynoir** [comment]
@Jarred-Sumner 

```sh
$ sudo npm -g install bun && bun -v && bun reproduce.mjs 

changed 3 packages, and audited 4 packages in 3s

found 0 vulnerabilities
0.5.9
1 | import chalk from 'chalk@5'
2 | 
3 | console.log(chalk.green('Hello!'))
               ^
TypeError: undefined is not an object (evaluating 'chalk.green')
      at /tmp/tmp.FlqjoULPvu/reproduce.mjs:3:12
```

**Comment by loynoir** [comment]
@Jarred-Sumner 

Emm. 

Sorry, seems something wrong on my side.

```sh
$ img="docker.io/archlinux:latest@sha256:f88544eecbe2b463e8e74c68241ef7ce0a5340fa93395abc05d4083036ece9cd"
$ docker run --rm -it -v $PWD:/app "$img" bash
[root ...]# pacman --noconfirm -Syu npm && npm -g install bun && bun -v && bun /app/reproduce.mjs
...
added 3 packages, and audited 4 packages in 40s

found 0 vulnerabilities
npm notice 
npm notice New major version of npm available! 8.19.2 -> 9.6.5
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.6.5
npm notice Run npm install -g npm@9.6.5 to update!
npm notice 
0.5.9
Hello!
```
